diff --git a/flutter/shell/platform/windows/egl/manager.cc b/flutter/shell/platform/windows/egl/manager.cc
--- a/flutter/shell/platform/windows/egl/manager.cc
+++ b/flutter/shell/platform/windows/egl/manager.cc
@@ -291,10 +291,20 @@ std::unique_ptr<WindowSurface> Manager::CreateWindowSurface(HWND hwnd,
                                        EGL_WIDTH,
                                        static_cast<EGLint>(width),
                                        EGL_HEIGHT,
                                        static_cast<EGLint>(height),
+                                       EGL_DIRECT_COMPOSITION_ANGLE,
+                                       EGL_TRUE,
                                        EGL_NONE};

  auto const surface = ::eglCreateWindowSurface(
      display_, config_, static_cast<EGLNativeWindowType>(hwnd),
      surface_attributes);
  if (surface == EGL_NO_SURFACE) {
    LogEGLError("Surface creation failed.");
    return nullptr;
  }

+  EGLint direct_composition = EGL_FALSE;
+  if (::eglQuerySurface(display_, surface, EGL_DIRECT_COMPOSITION_ANGLE,
+                        &direct_composition) != EGL_TRUE ||
+      direct_composition != EGL_TRUE) {
+    FML_LOG(ERROR) << "EGL DirectComposition surface verification failed.";
+    ::eglDestroySurface(display_, surface);
+    return nullptr;
+  }

  return std::make_unique<WindowSurface>(display_, render_context_->GetHandle(),
                                         surface, width, height);
 }
